### PR TITLE
Implement when clauses for debugger contributions

### LINKED ISSUE #133146: debugger contribution point should support a when clause
Just like `menus` and `taskDefinition` contribution points the `debugger` contribution point should also support `when` clauses. 

Here's where we handle the debugger extension point
https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts#L63

Here's where we define the schema https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/common/debugSchemas.ts#L14

Here's how Alex Ross does this for tasks 
https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts#L1663

I think the [debugAdapterManager](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts#L35) would have to evaluate all the `when` clauses when being asked for a list of debuggers and when setting the `debuggersAvailable` context key and we should be good.

@roblourens let me know if you are busy this milestone and I can also look into this
fyi @weinand

### PR DESCRIPTION
**Pull request by roblourens**
Fix #133146

I am fairly confident in this and tested it in normal OSS, but I don't have a perfect test setup for the web scenario with no debuggers enabled, do you have an idea @isidorn or if you think it looks good I can just check it in.

### CODE REVIEW

File: src/vs/workbench/contrib/debug/common/debugSchemas.ts [hunk 1]
```diff
@@ -67,6 +67,11 @@ export const debuggersExtPoint = extensionsRegistry.ExtensionsRegistry.registerE
 					description: nls.localize('vscode.extension.contributes.debuggers.configurationAttributes', "JSON schema configurations for validating \'launch.json\'."),
 					type: 'object'
 				},
+				when: {
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/common/debugSchemas.ts:70]: I would love if we added some examples in the description, similar to what AlexR did with tasks.
For example:
"Consider using `shellExecutionSupported`, `virtualWorkspace`, `resourceScheme` as appropriate for this"

**Review by isidorn — APPROVED** [review]
This looks great! Smoothly done.
I have just left one minor comment, but I can tackle it myself since it is just wording and will merge this in.
Great work :clap:

**Comment by isidorn** [comment]
Also I tried testing this out of source but it was hard to get the end to end flow. It looks good -> merging in.

**Comment by isidorn** [comment]
@roblourens ok, we do not really need a web scenario to test this out! We should just change the local `js-debug` extension when running out of source to add a `when` clause that is `false` and make sure that it is disabled when running locally out of source. I can try this out tomorrow.

**Comment by roblourens** [comment]
I did do that

**Pull request merged**